project/ottersome/README.md

```diff
@@ -0,0 +1,186 @@
+processing. This will be stated concretely and tackled generally. Once the
+course is over my aim is to use it for my own photography needs. The tools will 
+be more on the image editing/post-production side rather than something like
+openCV. A little disclaimer will be made: Due to restrictions in time I will
+most likely be able to implement only one or two of these features, but I state
+three in case time allows to cover them all. If I find that one of them doesn't
+interest me or is too lengthy then I might switch them as well.
+
+# Basic Information
+
+[Repository can be found here](https://github.com/ottersome/RIL).
+
+# Problem to Solve
+
+This is mostly a utility that solves problems of incovenience for myself. As
+I want a longer lasting project than a single semester I went with a broader
+subject.
+This is of use to anyone or any industry doing image processing. 
+For the sake of obtaining a sense of completion for the course, the three main goals
+will be chosen. Their inclusion is simply because they are a mix of previously acquired knowledge
+and experience I have been meaning to acquire
+
+1. [JPG Compression](https://people.cs.rutgers.edu/~elgammal/classes/cs334/slide9_short.pdf)
+   This uses a uses lossy compression through transform coding(DCT).
+   If time allows, I would also like to implement variations to see if there any
+   benefits to them.
+2. Kernel Application: A simple way of specifying your kernels and seeing their
+   final results on image or on application of image for the sake of easily
+   analyzing their effects without having to learn a complicated/bloated image
+   processing framework.
+3. Image Stitching to a Video: Like most people in a Lab I always work with
+   pyplot figures. I would like to be able to stitch them together into a video
+   without:
+
+        1. Having  a huge footprint on RAM 
+        2. The Final Video having a prohibitive size.
+
+# Prospective Users
+
+As this structure is quite general I would hope it could aid any photography or
+image processing aficionado. 
+
+It is in my hopes that I can later develop a simple GUI that would allow me to
+detach myself from software like Adobe Lightroom for my own photography needs.
+
+<!--- Most likely a dream lulz --->
+
+# System Architecture
+
+Results and input can be seen in the API section below. There is no one set work
+flow as this is a set of tools that can be used in any (reasonable) combination desired by
+the user. Constraints will come mainly from the I/O. I plan to stick to one raw
+data format + python buffers, anything else would likely not be implemented or
+imported from a different library once the class is over. This would be more
+about good efficient compression so images of absurd sizes +40Mb will not be
+considered within this semester. 
+
+
+## Interface
+
+### I/O Class
+
+This should handle all loading and saving operations.
+
+### Matrix Class
+
+A tensor buffer that will represent the image in memory. 
```

Comment:
Yes, you do need it.

FYI, take a look at https://github.com/solvcon/modmesh/blob/master/include/modmesh/SimpleArray.hpp or Eigen, etc.  My SimpleArray is much simpler as a starting point, and I can tell you everything about it.

Reply:
I will definitely have it as a reference. Thank you professor.